### DISCUSSION TITLE
How to create continuous map for large input files

### DISCUSSION THREAD

**Discussion by Udaykiran87** [stella-cv/stella_vslam#145] [Q&A] [answered]
Hello Community, 

Thanks for great work.

I have a huge chunk of rosbag files (consisting of approximately 15 files) of size approximately 30gb size.
Currently i am looking for non ros solution. I can convert rosbag files into mp4, but in order to make a complete map i need to combine all video files, but I think it will make the input file very very long. So I was thinking below options:

Option 1:
If can I join individual map files created from each video file using run_video_slam.cc? I guess it will be very very difficult to merge map files.

Option 2:
I was thinking to use run_video_localization.cc by passing inline parameter "--mapping", but i think it does not give me the cumulative or resultant map file (old map + map of new video) as output.

Could you please suggest me any good method to achieve above?

Thanks and Regards,
Udaykiran Patnaik.

**Comment by ymd-stella** [comment, marked as answer]
It would be easy to add an option to run_video_slam to load the map, or to run_video_localization to save the map.

**Reply by Udaykiran87** [reply]
Ok. Let me have a try.

**Reply by Udaykiran87** [reply]
I updated run_video_localization to save the map. Thank you for your suggestion.

**Comment by ymd-stella** [comment]
> I think it will make the input file very very long.

What is the problem with a large input file size?

**Reply by Udaykiran87** [reply]
There are some infrastructural constraints while combining all the videos.
